src/main/java/se/liu/ida/hefquin/engine/queryplan/ExecutablePlan.java

```diff
@@ -1,8 +1,18 @@
 package se.liu.ida.hefquin.engine.queryplan;
 
+import se.liu.ida.hefquin.engine.queryplan.executable.impl.pullbased.ResultElementIterator;
 import se.liu.ida.hefquin.engine.queryproc.QueryResultSink;
 
+import java.util.NoSuchElementException;
+
 public interface ExecutablePlan
 {
 	void run( QueryResultSink resultSink );
+
+    /**
+     * Returns the root operator of this plan.
+     */
+    ExecutableOperator getRootOperator();
+
+    ResultElementIterator getIterator();
```

Comment:
This needs to be moved to the PullBasedPlan class. There will also be push-based plans in the future and these will not be based on iterators but based on queues between the operators.